docs: `paths` and `patterns` difference in `no-restricted-imports` (#18273)
eslint/eslint: docs/src/rules/no-restricted-imports.md

This is also an object option whose value is an array. This option allows you to specify multiple modules to restrict using `gitignore`-style patterns.

Where `paths` option takes exact import paths, `patterns` option can be used to specify the import paths with more flexibility, allowing for the restriction of multiple modules within the same directory.

```json
"no-restricted-imports": ["error", {
  "paths": [{
    "name": "import-foo",
  }]
}]
```

> Here rule will only restrict the `import-foo` module.

Reviewer: ```suggestion
This configuration restricts import of the `import-foo` module but wouldn't restrict the import of `import-foo/bar` or `import-foo/baz`. You can use `patterns` to restrict both:
```

Author: Done!

> ```json
> "no-restricted-imports": ["error", {
>     "patterns": [{
>       "group": ["import1/*"],
>     }]
> }]
> ```
> 
> Here rule will restrict all modules inside `import1` directory.

Reviewer: ````suggestion
```json
"no-restricted-imports": ["error", {
    "paths": [{
      "name": "import-foo",
    }]
    "patterns": [{
      "group": ["import-foo/ba*"],
    }]
}]
```

This configuration restricts imports not just from `import-foo` using `path`, but also `import-foo/bar` and `import-foo/baz` using `patterns`.
````

Author: Done!

This is also an object option whose value is an array. This option allows you to specify multiple modules to restrict using `gitignore`-style patterns.

> Where `paths` option takes exact import paths, `patterns` option can be used to specify the import paths with more flexibility, allowing for the restriction of multiple modules within the same directory.

Author: Done! added examples.

> Where `paths` option takes exact import paths, `patterns` option can be used to specify the import paths with more flexibility, allowing for the restriction of multiple modules within the same directory.

Reviewer: ```suggestion
Where `paths` option takes exact import paths, `patterns` option can be used to specify the import paths with more flexibility, allowing for the restriction of multiple modules within the same directory. For example:
```
